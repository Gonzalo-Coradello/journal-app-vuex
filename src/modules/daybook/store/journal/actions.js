import journalApi from '@/api/journalApi'

export const loadEntries = async ({ commit }) => {
  const { data } = await journalApi.get('/entries.json')

  if (!data) {
    commit('setEntries', [])
    return
  }

  const entries = []

  for (let id of Object.keys(data)) {
    entries.push({
      id,
      ...data[id]
    })
  }

  commit('setEntries', entries)
}

export const updateEntry = async ({ commit }, entry) => {
  const { id, ...entryData } = entry
  await journalApi.put(`/entries/${id}.json`, entryData)

  commit('updateEntry', { ...entry })
}

export const createEntry = async ({ commit }, entry) => {
  const { data } = await journalApi.post(`/entries.json`, entry)

  commit('addEntry', { id: data.name, ...entry })
  return data.name
}

export const deleteEntry = async ({ commit }, id) => {
  console.log({ id })

  await journalApi.delete(`/entries/${id}.json`)

  commit('deleteEntry', id)
}
