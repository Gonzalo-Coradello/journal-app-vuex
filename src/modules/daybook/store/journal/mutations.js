// export const myMutation = (state) => {
//
// }

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries]
  state.isLoading = false
}

export const updateEntry = (state, entry) => {
  state.entries = state.entries.map((e) => (e.id === entry.id ? entry : e))
}

export const addEntry = (state, entry) => {
  state.entries.unshift(entry)
}

export const deleteEntry = (state, id) => {
  state.entries = state.entries.filter((entry) => entry.id !== id)
}
