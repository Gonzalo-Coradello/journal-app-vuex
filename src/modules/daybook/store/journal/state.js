export default () => ({
  isLoading: true,
  entries: [
    {
      id: '1',
      date: new Date().toDateString(),
      text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus rem at ducimus repellendus odio eos libero molestias officia illum temporibus commodi omnis, odit quas laudantium quo ipsa natus tempore magni.',
      picture: null,
    },
    {
      id: '2',
      date: new Date('2023/12/31').toDateString(),
      text: 'In, nobis similique! Voluptas, qui nobis maxime explicabo, laborum beatae consectetur laboriosam tempore dignissimos libero quas dolor voluptates odio temporibus? Voluptatum eligendi tempora nobis expedita qui provident perferendis quo reiciendis aperiam iure? Magni deleniti quaerat accusantium atque nisi!',
      picture: null,
    },
    {
      id: '3',
      date: new Date('2024/01/01').toDateString(),
      text: 'Repellendus consequatur atque accusamus iste vel sequi eveniet eos rem. Illum, ea? Rerum maxime quasi, quidem saepe temporibus facilis nesciunt sed nisi perspiciatis magni animi reprehenderit vitae error fuga, quaerat ad, excepturi assumenda quas autem enim!',
      picture: null,
    },
  ]
})
