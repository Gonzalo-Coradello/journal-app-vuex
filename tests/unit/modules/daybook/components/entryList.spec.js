import EntryList from '@/modules/daybook/components/EntryList.vue'
import { shallowMount } from '@vue/test-utils'
import { beforeEach, describe, expect, test, vi } from 'vitest'
import { createStore } from 'vuex'
import journal from '@/modules/daybook/store/journal'
import { journalState } from '../../../mock-data/test-journal-state'

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState }
      }
    }
  })

describe('Pruebas en el EntryList', () => {
  const store = createVuexStore(journalState)
  const mockRouter = {
    push: vi.fn()
  }

  let wrapper

  beforeEach(() => {
    vi.clearAllMocks()

    wrapper = shallowMount(EntryList, {
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [store]
      }
    })
  })

  test('debe llamar el getEntriesByTerm sin argumentos y mostrar 2 entradas', () => {
    expect(wrapper.findAll('entry-stub').length).toBe(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe llamar el getEntriesByTerm y filtrar las entradas', async () => {
    const input = wrapper.find('input')
    await input.setValue('aprender')

    expect(wrapper.findAll('entry-stub').length).toBe(1)
  })

  test('el boton de nueva entrada debe redireccionar a /new', () => {
    wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'entry', params: { id: 'new' } })
  })
})
