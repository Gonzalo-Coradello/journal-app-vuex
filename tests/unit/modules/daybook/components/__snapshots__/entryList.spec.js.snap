// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el EntryList > debe llamar el getEntriesByTerm sin argumentos y mostrar 2 entradas 1`] = `
"<div data-v-d2f66aca="" class="entry-list-container">
  <div data-v-d2f66aca="" class="px-2 pt-2"><input data-v-d2f66aca="" type="text" placeholder="Buscar entrada" class="form-control"></div>
  <div data-v-d2f66aca="" class="mt-2 d-flex flex-column"><button data-v-d2f66aca="" class="btn btn-primary mx-3"><i data-v-d2f66aca="" class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div data-v-d2f66aca="" class="entry-scrollarea">
    <entry-stub data-v-d2f66aca="" entry="[object Object]"></entry-stub>
    <entry-stub data-v-d2f66aca="" entry="[object Object]"></entry-stub>
  </div>
</div>"
`;
